### PR TITLE
feat(rs-dpp): backports of identity/stateTransition from js-dpp

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

PR ensures that identity/stateTransition logic in rs-dpp matches js-dpp

## What was done?

- Provided executionContext to mark_asset_lock_transaction_as_used
- Fetch identity balance instead of identity in identity create transition state validator

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-dpp/src/identity/state_transition/identity_create_transition/validation/state/mod.rs [hunk 1]
```diff
@@ -20,19 +19,16 @@ pub async fn validate_identity_create_transition_state(
...
+        .fetch_identity_balance(identity_id, state_transition.get_execution_context())
+        .await
         .map_err(|e| NonConsensusError::StateRepositoryFetchError(e.to_string()))?;
 
     if state_transition.get_execution_context().is_dry_run() {
         return Ok(result);
     }
 
-    if let Some(_identity) = maybe_identity {
+    if let Some(_balance) = balance {
```
Comment (shumkov) [inline packages/rs-dpp/src/identity/state_transition/identity_create_transition/validation/state/mod.rs:31]: balance.is_some()

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-dpp/src/identity/state_transition/identity_create_transition/validation/state/mod.rs (hunk 1 above)
Comment (markin-io) [reply packages/rs-dpp/src/identity/state_transition/identity_create_transition/validation/state/mod.rs]: Fixed

**Review by antouhou — APPROVED** [review]
👍

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**